agent/src/main/java/com/code_intelligence/jazzer/api/Jazzer.java

```diff
@@ -23,68 +23,83 @@
  * Helper class with static methods that interact with Jazzer at runtime.
  */
 final public class Jazzer {
-  private static Class<?> jazzerInternal = null;
+  private static final Class<?> jazzerInternal;
 
-  private static MethodHandle traceStrcmp = null;
-  private static MethodHandle traceStrstr = null;
-  private static MethodHandle traceMemcmp = null;
+  private static final MethodHandle traceStrcmp;
+  private static final MethodHandle traceStrstr;
+  private static final MethodHandle traceMemcmp;
 
-  private static MethodHandle consume = null;
-  private static MethodHandle autofuzzFunction1 = null;
-  private static MethodHandle autofuzzFunction2 = null;
-  private static MethodHandle autofuzzFunction3 = null;
-  private static MethodHandle autofuzzFunction4 = null;
-  private static MethodHandle autofuzzFunction5 = null;
-  private static MethodHandle autofuzzConsumer1 = null;
-  private static MethodHandle autofuzzConsumer2 = null;
-  private static MethodHandle autofuzzConsumer3 = null;
-  private static MethodHandle autofuzzConsumer4 = null;
-  private static MethodHandle autofuzzConsumer5 = null;
+  private static final MethodHandle consume;
+  private static final MethodHandle autofuzzFunction1;
+  private static final MethodHandle autofuzzFunction2;
+  private static final MethodHandle autofuzzFunction3;
+  private static final MethodHandle autofuzzFunction4;
+  private static final MethodHandle autofuzzFunction5;
+  private static final MethodHandle autofuzzConsumer1;
+  private static final MethodHandle autofuzzConsumer2;
+  private static final MethodHandle autofuzzConsumer3;
+  private static final MethodHandle autofuzzConsumer4;
+  private static final MethodHandle autofuzzConsumer5;
 
   static {
+    Class<?> jazzerInternalLocal = null;
+    MethodHandle traceStrcmpLocal = null;
+    MethodHandle traceStrstrLocal = null;
+    MethodHandle traceMemcmpLocal = null;
+    MethodHandle consumeLocal = null;
+    MethodHandle autofuzzFunction1Local = null;
+    MethodHandle autofuzzFunction2Local = null;
+    MethodHandle autofuzzFunction3Local = null;
+    MethodHandle autofuzzFunction4Local = null;
+    MethodHandle autofuzzFunction5Local = null;
+    MethodHandle autofuzzConsumer1Local = null;
+    MethodHandle autofuzzConsumer2Local = null;
+    MethodHandle autofuzzConsumer3Local = null;
+    MethodHandle autofuzzConsumer4Local = null;
+    MethodHandle autofuzzConsumer5Local = null;
```

Comment:
Could you clarify why we need the temporary variables? IISC, they are assigned once and we could therefore use the final fields directly

Reply:
We unfortunately can't due to the checked exceptions: All finals have to end up initialized in all code paths, including if an exception is thrown in the middle. The only alternative would be to extract the individual initializations into functions.

Reply:
I see. Thanks!